crates/nu-command/src/filesystem/ls.rs

```diff
@@ -90,6 +90,13 @@ impl Command for Ls {
         let pattern_arg = opt_for_glob_pattern(engine_state, stack, call, 0)?;
         let pattern_arg = {
             if let Some(path) = pattern_arg {
+                // it makes no sense to list an empty string.
+                if path.item.as_ref().is_empty() {
+                    return Err(ShellError::FileNotFoundCustom {
+                        msg: "empty string('') file does not exist".to_string(),
```

Comment:
would it be more likely that you expect to use expect to use it for a directory?
```suggestion
                        msg: "empty string('') directory or file does not exist".to_string(),
```

I don't know my platforms well enough if any system allows an empty string file/directory name at all.

Reply:
I've tried on both linux/mac/windows, and they don't allow empty string file, with different error message.

Reply:
OK perfect! Let's land this :)